### PR TITLE
[rails__actioncable] Create types

### LINKED ISSUE #51455: @types/actioncable package
- [ x] I tried using the `@types/xxxx` package and had problems.
- Authors: @zhu1230 @szechyjs 

I tried to pull in type declarations for ActionCable, but due to the fact that I am using the package `@rails/actioncable` the declarations would not resolve and I got the following in my text editor:
```
Could not find a declaration file for module '@rails/actioncable'. '... /node_modules/@rails/actioncable/app/assets/javascripts/action_cable.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/rails__actioncable` if it exists or add a new declaration (.d.ts) file containing `declare module '@rails/actioncable';`ts(7016)
```

I copied the declaration into my own types folder, and changed [this line](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/actioncable/index.d.ts#L50) to 

```
declare module '@rails/actioncable' {
```

Adding the `@rails/` prefix seems to have fixed it. There are two `actioncable` npm packages, and it looks to me like `@rails/actioncable` is the canonical version.

Is there any way I can resolve this naming in my tsconfig? Or should this package be updated with my change above?

### PR DESCRIPTION
**Pull request by martin-badin**
Closes: #51455

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by martin-badin** [comment]
@pesoklp13 Please, could you review it?

File: types/rails__actioncable/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+ * @see https://github.com/rails/rails/blob/main/actioncable/app/javascript/action_cable/subscription.js
+ */
+export class Subscription {
+    constructor(consumer: Consumer, params?: object, mixin?: any);
+
+    perform(action: string, data?: object): boolean;
+
+    send(data: object): boolean;
+
+    unsubscribe(): this;
```
Comment (pesoklp13) [inline types/rails__actioncable/index.d.ts:123]: interesting notation

**Review by pesoklp13 — APPROVED** [review]

File: types/rails__actioncable/index.d.ts (hunk 1 above)
Comment (martin-badin) [reply types/rails__actioncable/index.d.ts:123]: The function returns the same object as you want to unsubscribe

Comment (pesoklp13) [reply types/rails__actioncable/index.d.ts:123]: Yeah, I got it, but didn't see it in action before :D

**Pull request merged**